core/src/avm2/bytearray.rs

```diff
@@ -23,6 +25,17 @@ pub enum CompressionAlgorithm {
     Lzma,
 }
 
+pub struct EofError;
+
+impl EofError {
+    pub fn to_avm<'gc>(&self, activation: &mut Activation<'_, 'gc>) -> Error<'gc> {
```

Comment:
let's make this `#[inline(never)]`.
TBH I wish we didn't need to sprinkle this in every method. In a way it feels more "proper" for the Bytearray object itself to "throw the exception", instead of doing it in the binding. But both ways have their downsides :/

Reply:
Done!